Add opentracing operation name settings

JorritSalverda opened the pull request:
**What this PR does / why we need it**:

In order to set custom span names for server and location span use the following config items in the configmap:

```
opentracing-operation-name: "HTTP $request_method $uri"
opentracing-location-operation-name: "HTTP $request_method $uri"
```

**Which issue this PR fixes**:

fixes #4116

JorritSalverda commented:
/assign @aledbf

ElvinEfendi commented:
@JorritSalverda thanks for the PR, please write e2e tests to cover these new functionality.

JorritSalverda commented:
@ElvinEfendi I've added tests confirming presence of the directives in the generated config. I guess it's easiest to add tests for https://github.com/kubernetes/ingress-nginx/pull/4117 when this one's merged.

aledbf commented:
/ok-to-test

aledbf commented:
@JorritSalverda please rebase and squash the commits

JorritSalverda commented:
@aledbf I've squashed and rebased. I'm not fully clear on why the tests have failed, whether it's an error in my tests or the test run experiences timeouts for another reason.

aledbf commented:
> I'm not fully clear on why the tests have failed, whether it's an error in my tests or the test run experiences timeouts for another reason

The errors are related to your tests. (you can check the failures https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/ingress-nginx/4118/pull-ingress-nginx-e2e-1-13/1136971198354165760)

aledbf commented:
@JorritSalverda friendly ping

aledbf commented:
@JorritSalverda please reopen when you can fix the e2e issues
